src/main/common/sensor_alignment.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "stdbool.h"
+
+#include "platform.h"
+
+#include "common/sensor_alignment.h"
+
+
+void buildRotationMatrixFromAlignment(const sensorAlignment_t* sensorAlignment, fp_rotationMatrix_t* rm)
+{
+    fp_angles_t rotationAngles;
+    rotationAngles.angles.roll  = DEGREES_TO_RADIANS(sensorAlignment->values.roll);
+    rotationAngles.angles.pitch = DEGREES_TO_RADIANS(sensorAlignment->values.pitch);
+    rotationAngles.angles.yaw   = DEGREES_TO_RADIANS(sensorAlignment->values.yaw);
+
+    buildRotationMatrix(&rotationAngles, rm);
+}
+
+bool isSensorAlignmentStandard(sensorAlignment_t* sensorAlignment)
+{
+    bool nonStandardAlignmentDetected = false;
+
+    for (int i = 0; i < XYZ_AXIS_COUNT && !nonStandardAlignmentDetected; i++) {
+        nonStandardAlignmentDetected = (int)sensorAlignment->raw[i] % 90 != 0;
+    }
+
+    return !nonStandardAlignmentDetected;
+}
+
+void updateStandardAlignmentFromNonDefaultLegacyRotation(sensorAlignment_t* sensorAlignment, sensor_align_e rotation)
+{
+    if (rotation == ALIGN_DEFAULT) {
+        return;
+    }
+
+    if (!isSensorAlignmentStandard(sensorAlignment)) {
+        return;
+    }
+
+    buildAlignmentFromLegacyRotation(sensorAlignment, rotation);
+}
+
+void buildAlignmentFromLegacyRotation(sensorAlignment_t* sensorAlignment, sensor_align_e rotation)
+{
+    *sensorAlignment = CUSTOM_ALIGN_CW0_DEG;
+
+    switch (rotation) {
+    default:
+    case CW0_DEG:
```

Comment:
This can be implemented as array (store yaw/pitch divided by 90), possibly packing it 2 bits per euler angle

Reply:
this code is now currently only test code.  if the code gets re-used in the firmware I may investigate this further.